server/jamovi/tests/test_dataset.py

```diff
@@ -48,6 +48,11 @@ def test_text(empty_dataset):
     column.set_value(2, "jim")
     column.set_value(3, "33.1")
 
+    # TEMP: commit transaction - pending design descision/clarification - should be automatic?
+    commit_method = getattr(ds, "commit_set_values")
```

Comment:
so we could use this test just with `DuckDataset`, or we could use it with *both* `DuckDataset` and `ShMemDataset` ...

if it's the former, you can just go `ds.commit_set_values()`, if it's the latter, you'll need to handle the `AttributeNotFoundError` (or whatever it is) when `ShMemDataset` doesn't have a `commit_set_values()` method.

(we can just add stubs to `ShMemDataset` and call it a day ... but that's cython that needs to be rebuilt ... so this is easier for now).

as for automatic committing ... yup, commit should occur automatically under a lot of scenarios.

a call to `get_value()` on `DuckDataset` for example should trigger a commit. any changes to column attributes, changes to the row count, etc. etc. if you look in `DuckDataset` you'll find a lot of `data_cache.clear()`s ... we'll almost need as many `write_buffer.commit()`s

`DuckDataset`'s `detach()` is another one ... that's when the file is being closed ... so we want to flush everything before that happens.